server_test.go

```diff
@@ -208,6 +208,51 @@ func TestOpenStatRace(t *testing.T) {
 	checkServerAllocator(t, server)
 }
 
+func TestOpenWithPermissions(t *testing.T) {
+	client, server := clientServerPair(t)
+	defer client.Close()
+	defer server.Close()
+
+	tmppath := path.Join(os.TempDir(), "open_permissions")
+	pflags := flags(os.O_RDWR | os.O_CREATE | os.O_TRUNC)
+	ch := make(chan result, 2)
+	id1 := client.nextID()
+	id2 := client.nextID()
+
+	// New files should have their permissions set.
+	client.dispatchRequest(ch, &sshFxpOpenPacket{
+		ID:     id1,
+		Path:   tmppath,
+		Pflags: pflags,
+		Flags:  sshFileXferAttrPermissions,
+		Attrs:  []byte{0x0, 0x0, 0x1, 0xe5}, // 0o745 -- a slightly strange permission to test.
+	})
+	<-ch
```

Comment:
Should probably test the error returned from the `dispatchRequest` channel return?

Why not actually just avoid channel tedium here entirely and use the slightly higher level `sendPacket` which will do the receive and extract the `error` from the `result` into a return value for you?